### ISSUE TITLE
Test for reconnection are randomly breaking

### ISSUE THREAD

**Issue by cheshir** [cheshir/go-mq#25]
**STR**
Run: `go test -race -mock-broker=1`

Expected result:
```
PASS
ok  	github.com/cheshir/go-mq	2.318s
```

Actual result:
```
--- FAIL: TestMq_Reconnect (0.32s)
	mq_test.go:294: Consumer did not read messages. Produced 2, read 0
```

Notes: 
Test works only with mocked broker.
Can't reproduce bug with enabled debugger or with `-run=Reconnect` flag.

**Comment by cheshir** [comment]
Still failing.

**Comment by cheshir** [comment]
Some fixes was made. Now I can't reproduce this issue locally.
Seems like this problem doesn't affect work with real broker and. Probably, the cause of the issue is inside the amqptest package.

Need help.
Test skipped in v1.0.1 until fix.

**Comment by cheshir** [comment]
The problem can be reproduced only in environment where only CPU is available (build machine).
Fixed in #29.
